src/libs/DebugUtils.ts

```diff
@@ -551,6 +612,164 @@ function validateReportActionJSON(json: string) {
     });
 }
 
+/**
+ * Gets the reason for showing LHN row
+ *
+ * @param report
+ * @returns translation key or null
+ */
+function getReasonForShowingRowInLHN(report: OnyxEntry<Report>): TranslationPaths | null {
+    if (!report) {
+        return null;
+    }
+
+    const isInDefaultMode = !isInFocusMode;
+
+    if (hasValidDraftComment(report.reportID)) {
+        return 'debug.reasonVisibleInLHN.hasDraftComment';
+    }
+
+    if (ReportUtils.requiresAttentionFromCurrentUser(report)) {
+        return 'debug.reasonVisibleInLHN.hasGBR';
+    }
+
+    if (report.isPinned) {
+        return 'debug.reasonVisibleInLHN.pinnedByUser';
+    }
+
+    const reportIsSettled = report.statusNum === CONST.REPORT.STATUS_NUM.REIMBURSED;
+
+    if (ReportUtils.isExpenseRequest(report) && !reportIsSettled) {
+        return 'debug.reasonVisibleInLHN.isNonReimbursedIOU';
+    }
+
+    if (report.errorFields?.addWorkspaceRoom) {
+        return 'debug.reasonVisibleInLHN.hasAddWorkspaceRoomErrors';
+    }
+
+    if (isInFocusMode && ReportUtils.isUnread(report) && ReportUtils.getReportNotificationPreference(report) !== CONST.REPORT.NOTIFICATION_PREFERENCE.MUTE) {
+        return 'debug.reasonVisibleInLHN.isUnread';
+    }
+
+    if (isInDefaultMode && ReportUtils.isArchivedRoom(report, ReportUtils.getReportNameValuePairs(report?.reportID))) {
+        return 'debug.reasonVisibleInLHN.isArchived';
+    }
+
+    if (ReportUtils.isSelfDM(report)) {
+        return 'debug.reasonVisibleInLHN.isSelfDM';
+    }
+
+    return 'debug.reasonVisibleInLHN.isFocused';
+}
+
+/**
+ * Gets the reason that is causing the GBR to show up in LHN row
+ *
+ * @param report
+ * @param parentReportAction
+ * @returns translation key or null
+ */
+function getReasonForShowingGreenDotInLHNRow(report: OnyxEntry<Report>, parentReportAction?: ReportAction): TranslationPaths | null {
```

Comment:
```suggestion
/**
 * Gets the reason that is causing the GBR to show up in LHN row.
 */
function getReasonForShowingGreenDotInLHNRow(report: OnyxEntry<Report>, parentReportAction?: ReportAction): TranslationPaths | null {
```

Could you also add a comment in the function about where you based this logic?